gpio/AM2302_sensor/AM2302-Sensor.cpp

```diff
@@ -0,0 +1,252 @@
+/*
+*	AM2302-Sensor.cpp
+*
+*	Author: Frank Häfele
+*	Date:	21.11.2023
+*
+*	Object: Measure Sensor Data of AM2302-Sensor
+*
+*/
+#include "AM2302-Sensor.hpp"
+
+/**
+ * @brief Construct a new am2302::am2302 sensor::am2302 sensor object
+ * 
+ * @param pin Pin for AM2302 sensor
+ */
+AM2302::AM2302_Sensor::AM2302_Sensor(uint8_t pin) : _millis_last_read{0}, _pin{pin}
+{}
+
+/**
+ * @brief begin function setup pin and run sensor check.
+ * 
+ * @return true if sensor check is successful.
+ * @return false if sensor check failed.
+ */
+bool AM2302::AM2302_Sensor::begin() {
+   gpio_init(_pin);
+   gpio_pull_up(_pin);
+   // required delay() for a secure sensor check,
+   // if you reset the mcu very fast one after another
+   auto tic{time_us_64()};
+   while ( time_us_64() - tic < READ_FREQUENCY * 1000U ) {
+      sleep_ms(1U);
+   }
+   auto status{read()};
+   _millis_last_read = time_us_64();
```

Comment:
`time_us_64` returns the time in microseconds https://www.raspberrypi.com/documentation/pico-sdk/hardware.html#ga7bbcd996dbd3dd4adc03a80272fb2fc3 so calling the variable `_millis_last_read` might be confusing to whoever reads this code in future? :wink:

Reply:
sure youre right,
I should adapt this
👍